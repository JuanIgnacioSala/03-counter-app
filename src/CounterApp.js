// import React, { Fragment } from "react";
import React, {useState} from "react";
import PropTypes from 'prop-types';

const CounterApp = ({value = 10}) =>{


    const [counter, setCounter] = useState( value );
    
    //handleAdd
    const handleAdd = ()=> {
        setCounter(counter + 1);
        //setCounter((c)=> c + 1);
    }
    const handleReset = ()=> {
        setCounter(value);
        //setCounter((c)=> c + 1);
    }
    const handleSubstra = ()=> {
        setCounter(counter - 1);
        //setCounter((c)=> c + 1);
    }

    return (<>
    <h1> Contador </h1>
    <h2>{counter}</h2>

    <button  className='but' onClick={ handleSubstra }>-1</button>
    <button onClick={ handleReset }>Reset</button>
    <button className='but' onClick={ handleAdd }>+1</button>
    </>)
}
CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;
