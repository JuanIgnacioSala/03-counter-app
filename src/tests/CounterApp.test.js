import { getByText, render } from "@testing-library/react"
import React from 'react';
import {shallow} from 'enzyme'
import CounterApp from "../CounterApp";

describe('Pruebas en CounterApp',()=>{

    let wrapper = shallow(<CounterApp/>);

    beforeEach(()=>{

        wrapper = shallow(<CounterApp/>);
    })

    test('Debe mostrar <CounterApp/> correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Mostrar valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={100}/>)
        const valorContador = wrapper.find('h2').text();
        expect(valorContador).toBe('100');
    })
    
    test('incrementar con el boton +1 ', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('11')
    })
    
    test('Restar con el boton -1 ', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('9')
    })
    
    test('Reset counter ', () => {

        const wrapper = shallow(<CounterApp value={100}/>)
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text();
       expect(counterText).toBe('100');
    })
     
})