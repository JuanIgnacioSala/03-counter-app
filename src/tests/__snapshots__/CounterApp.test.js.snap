// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe mostrar <CounterApp/> correctamente  1`] = `
<Fragment>
  <h1>
     Contador 
  </h1>
  <h2>
    10
  </h2>
  <button
    className="but"
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="but"
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
